apps/src/javalab/javalabRedux.js

```diff
@@ -379,5 +414,38 @@ export default function reducer(state = initialState, action) {
       isVisualizationCollapsed: !state.isVisualizationCollapsed
     };
   }
+  if (action.type === ADD_PLAYGROUND_ITEM_DATA) {
```

Comment:
Not sure how difficult this would be but do you think it's worth pulling out Playground specific actions into it's own reducer? Especially if we start using redux for other Javalab miniapps/components, it might be nice to isolate each to it's own reducer. I'm fine with keeping these here for now though since it's only a few actions.

Reply:
that might be a good call. I initially didn't because it was a few items, but it ended up being 4 actions so it adds a few lines of code

Reply:
done!